figtree.go

```diff
@@ -1003,19 +1003,6 @@ func (m *Merger) assignValue(dest reflect.Value, src mergeSource, opts assignOpt
 		return true, nil
 	}
 
-	// if we have a collection don't proceed to attempt to unmarshal direct
-	// from the yaml.Node ... collections are process per item, rather than
-	// as a whole.
-	if isCollection(dest) {
```

Comment:
This moved below the isSpecial/CanAddr test so we can check to see if the structure has an UnmarhalYAML func defined even if it is a "collection".